src/java/org/apache/cassandra/serializers/MapSerializer.java

```diff
@@ -21,6 +21,7 @@
 import java.nio.BufferUnderflowException;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
+import java.util.Collection;
```

Comment:
I did not change that code. So, not sure what you are referring to.

Reply:
That second check looks like dead code that was forgotten when fixing the validation method during CASSANDRA-18504. There is an identical situation in `SetSerializer`. 

This is unrelated to the changes on this PR but, since we are here, we could just remove the two `if (accessor.isEmpty(input)) return;` lines. This would save us the overhead of a separate trivial ticket, if that's ok with you.

Reply:
Thanks for confirming it is dead code @adelapena and not a bug; I leave it to @blerer to decide whether he wants to clear that, too, or leave it for another ticket. The patch is big already, I understand.

Reply:
I will remove them.